### DISCUSSION TITLE
How to use event tracking to modify credits balance

### DISCUSSION THREAD

**Discussion by chendwww** [uselotus/lotus#669] [Q&A] [answered]
Hello, We're new to Lotus. We want to use event tracking modify credits directly.

We find there is an advance setting:

"Does this metric represent a cost?" , if we turn it on will it reduce total credits of the customers directly? If not, how can we modify creidts directly by events?

**Comment by chendwww** [comment, marked as answer]
We provide more details about our needs.

We have a balance system, our customers pay $1000 to us and we will add $1000 to their balance. 

We have an international text message system, we charge customers by text message cost ( different countries have different cost per text,  and different types of messages, for example, text message to China is 0.04,  voice message to US is 0.1 and so on).

We use an unquie propety "cost" in event tracking.  we will calculate cost and send it to cost propety, just as cost: 0.04.

We want to reduce the customer balance directly by the cost propety in event tracking. How to do it?

**Reply by mnida** [reply]
Hi @chendwww . I think I caught up with you in Slack as well. However, after seeing this description, we don't yet support instant balance subtractions. Right now you would do this by creating a plan and checking the draft invoice of the plan to see how much cost will show up on the invoice at the end of the month. Then once the invoice is issued, it will deduct from the balance but not during the draft invoice period.

However, this has been noted and we have started working on a fix for this on our near-term roadmap. If interested feel free to follow up on our Slack.

**Reply by Dima1506** [reply]
Hi, is the project supported? will there be updates and how can it get into the slack?
